From f4d9be9930865f295c07707edf712da742069000 Mon Sep 17 00:00:00 2001
From: Lin Ruizhe <linruizhe@huawei.com>
Date: Fri, 31 Dec 2021 15:59:24 +0800
Subject: [PATCH] irq_desc: Add kabi_reserve in irq_desc
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f4d9be9930865f295c07707edf712da742069000


hulk inclusion
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MZU1
CVE: NA

---------------------------

Add KABI_RESERVE in irq_desc

Signed-off-by: Lin Ruizhe <linruizhe@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/irqdesc.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/irqdesc.h b/include/linux/irqdesc.h
index 5745491303e0..dc1b213ae941 100644
--- a/include/linux/irqdesc.h
+++ b/include/linux/irqdesc.h
@@ -5,7 +5,7 @@
 #include <linux/rcupdate.h>
 #include <linux/kobject.h>
 #include <linux/mutex.h>
-
+#include <linux/kabi.h>
 /*
  * Core internal functions to deal with irq descriptors
  */
@@ -102,6 +102,7 @@ struct irq_desc {
 	int			parent_irq;
 	struct module		*owner;
 	const char		*name;
+	KABI_RESERVE(1)
 } ____cacheline_internodealigned_in_smp;
 
 #ifdef CONFIG_SPARSE_IRQ
-- 
2.26.2

